### PR TITLE
Get last tx after restart & chain txs on that utxo

### LINKED ISSUE #537: Bitcoin DA: get last tx after restart
Rn we use utxo as inputs for a new tx. We should be able to build a chain of utxos to get the last tx and attach the first sequencer tx on DA to it.

### PR DESCRIPTION
**Pull request by ercecan**
# Description
Gets last tx after restart & chain txs on that utxo

## Linked Issues
- Fixes #537

### CODE REVIEW

File: crates/bitcoin-da/src/service.rs [hunk 1]
```diff
@@ -545,6 +589,34 @@ impl DaService for BitcoinService {
...
+                if inscription.get_sig_verified_hash().is_some() {
+                    // Decompress the blob
+                    let decompressed_blob = decompress_blob(&inscription.body);
+
+                    let relevant_tx = BlobWithSender::new(
+                        decompressed_blob,
+                        inscription.public_key,
+                        sha256d::Hash::hash(&inscription.body).to_byte_array(),
+                    );
+
```
Comment (eyusufatik) [inline crates/bitcoin-da/src/service.rs:613]: we do this in another function as well. make a shared function among these two.

**Review by eyusufatik — CHANGES_REQUESTED** [review]

**Review by eyusufatik — APPROVED** [review]

File: crates/bitcoin-da/src/service.rs [hunk 2]
```diff
@@ -98,7 +99,21 @@ impl BitcoinService {
             tokio::runtime::Handle::current().block_on(async move {
                 // TODO https://github.com/chainwayxyz/citrea/issues/537
                 // TODO find last tx by utxo chain
-                let mut prev_tx = None;
+                let mut prev_tx = match this.get_pending_transactions().await {
+                    Ok(pending_txs) => {
+                        if !pending_txs.is_empty() {
+                            let tx = pending_txs.first().unwrap().clone();
```
Comment (yaziciahmet) [inline crates/bitcoin-da/src/service.rs:105]: is pending_txs guaranteed to be ordered?

Comment (kpp) [inline crates/bitcoin-da/src/service.rs:105]: The logic is unclear to me. Shouldnt we sort by height desc?

File: crates/bitcoin-da/src/spec/transaction.rs [hunk 3]
```diff
@@ -24,6 +24,10 @@ impl TransactionWrapper {
         };
         Self { tx }
     }
+
+    pub fn inner(&self) -> &BitcoinTransaction {
```
Comment (kpp) [inline crates/bitcoin-da/src/spec/transaction.rs:28]: No need, there is Deref trait

**Review by kpp — APPROVED** [review]

**Pull request merged**